Source/Caches/HACachedStates.swift

```diff
@@ -6,26 +6,24 @@ public extension HACachesContainer {
 /// Cached version of all entity states
 public struct HACachedStates {
     /// All entities
-    public var all: Set<HAEntity>
+    public var all: Set<HAEntity> = []
     /// All entities, keyed by their entityId
     public subscript(entityID: String) -> HAEntity? {
         get { allByEntityId[entityID] }
         set { allByEntityId[entityID] = newValue }
     }
 
     /// Backing dictionary, whose mutation updates the set
-    private var allByEntityId: [String: HAEntity] {
+    public var allByEntityId: [String: HAEntity] {
```

Comment:
I think this should be 'internal' more than 'public'. The struct itself acts as the subscript that should be used to access this.